cmd/juju/waitfor/unit.go

```diff
@@ -230,6 +230,11 @@ func (m UnitScope) GetIdents() []string {
 func (m UnitScope) GetIdentValue(name string) (query.Box, error) {
 	m.ctx.RecordIdent(name)
 
+	// UnitInfo could be nil - handle it here to avoid nil pointer dereference
+	if m.UnitInfo == nil {
+		return query.NewString("unknown"), nil
```

Comment:
I'll leave this for @SimonRichardson , but I think this "unknown" value was for when the "*-status" values are requested. Other values I think empty string is probably valid.  I think it would be better to understand how we are getting into this state.

Reply:
Just return the error. If unit info is nil, something is broken elsewhere and we're just papering up the cracks.

Reply:
yeah you're right, we should return an error